File: DuasTelas/app/src/main/java/com/example/duastelas/MainActivity.java
package com.example.duastelas;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void Abrir(View view) {
        Toast.makeText(this,"Abrir OK",Toast.LENGTH_LONG).show();
        Intent i = new Intent(MainActivity.this, SegundaActivity.class);
        startActivity(i);
    }

    public void message(View view) {
        EditText edtName = findViewById(R.id.editName);
        String name = edtName.getText().toString();
        Toast.makeText(this, "Boa Noite " + name, Toast.LENGTH_SHORT).show();
    }

    public void mudar(View view) {
        EditText edtName = findViewById(R.id.editName);
        edtName.setText("mudou");

    }

    public void quadrado(View view) {
        EditText edtName = findViewById(R.id.editName);
        float numero = 0;
        try {
            numero = Float.parseFloat(edtName.getText().toString());
        }
        catch (Exception e){
            Toast.makeText(this, e.getMessage(),Toast.LENGTH_LONG).show();
        }
        numero = numero * numero;
        if(numero>0){
            edtName.setText(String.valueOf(numero));
        }


    }
}

File: 14Biometria/app/src/main/java/com/example/a14biometria/MainActivity.kt
package com.example.a14biometria

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.biometric.BiometricPrompt
import androidx.core.content.ContextCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var executor = ContextCompat.getMainExecutor(this)
        var biometricPrompt = BiometricPrompt(this, executor,
            object : BiometricPrompt.AuthenticationCallback() {
                override fun onAuthenticationError(errorCode: Int,
                                                   errString: CharSequence) {
                    super.onAuthenticationError(errorCode, errString)
                    Toast.makeText(applicationContext,
                        "Authentication error: $errString", Toast.LENGTH_SHORT)
                        .show()
                }

                override fun onAuthenticationSucceeded(
                    result: BiometricPrompt.AuthenticationResult) {
                    super.onAuthenticationSucceeded(result)
                    Toast.makeText(applicationContext,
                        "Authentication succeeded!", Toast.LENGTH_SHORT)
                        .show()
                    var tx = findViewById<TextView>(R.id.textview)
                    tx.setText("Autenticado!")
                }

                override fun onAuthenticationFailed() {
                    super.onAuthenticationFailed()
                    Toast.makeText(applicationContext, "Authentication failed",
                        Toast.LENGTH_SHORT)
                        .show()
                }
            })

        var promptInfo = BiometricPrompt.PromptInfo.Builder()
            .setTitle("Biometric login for my app")
            .setSubtitle("Log in using your biometric credential")
            .setNegativeButtonText("Use account password")
            .build()

        // Prompt appears when user clicks "Log in".
        // Consider integrating with the keystore to unlock cryptographic operations,
        // if needed by your app.
        val biometricLoginButton =
            findViewById<Button>(R.id.biometric_login)
        biometricLoginButton.setOnClickListener {
            biometricPrompt.authenticate(promptInfo)
        }
    }
/*
    fun verificar(){
        val executor = ContextCompat.getMainExecutor(this)
        val biometricManager = androidx.biometric.BiometricManager.from(this)
        when (biometricManager.canAuthenticate()) {
            BiometricManager.BIOMETRIC_SUCCESS ->
                autenticar(executor)
            BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE ->
                Toast.makeText(
                    this,

                    "Erro no Leitor biométrico",

                    Toast.LENGTH_LONG
                ).show()
            BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE ->
                Toast.makeText(
                    this,

                    "Leitor biométrico não disponível",

                    Toast.LENGTH_LONG
                ).show()
            BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED ->
                Toast.makeText(
                    this,

                    "Erro: nenhuma digital cadastrada",

                    Toast.LENGTH_LONG
                ).show()
        }
    }

    private fun autenticar(executor: Executor) {
        val promptInfo = BiometricPrompt.PromptInfo.Builder()
            .setTitle("Acesso com a digital")
            .setSubtitle("subtitulo")
            .setDescription("Posicione o dedo no leitor biométrico do seu celular")
            .setDeviceCredentialAllowed(true)
            .build()
        val biometricPrompt = BiometricPrompt(this, executor,
            object : BiometricPrompt.AuthenticationCallback() {
                override fun onAuthenticationSucceeded(
                    result: BiometricPrompt.AuthenticationResult
                ) {
                    super.onAuthenticationSucceeded(result)
//main_layout.visibility = View.VISIBLE
                    Toast.makeText(applicationContext,"LOGIN OK",

                        Toast.LENGTH_LONG).show()
                }
                override fun onAuthenticationError(
                    errorCode: Int, errString: CharSequence
                ) {
                    super.onAuthenticationError(errorCode, errString)
                    Toast.makeText(
                        applicationContext,
                        "Erro:" + errString,
                        Toast.LENGTH_SHORT
                    ).show()
                }
                override fun onAuthenticationFailed() {
                    super.onAuthenticationFailed()
                    Toast.makeText(applicationContext,
                        " Erro: Autenticação falhou.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            })
        biometricPrompt.authenticate(promptInfo)
    }*/
}